logging/logger_manager.go

```diff
@@ -0,0 +1,43 @@
+package logging
+
+import (
+	"fmt"
+	"sync"
+)
+
+type loggerRegistry struct {
+	mu      sync.RWMutex
+	loggers map[string]Logger
+}
+
+var loggerManager = newLoggerManager()
+
+func newLoggerManager() *loggerRegistry {
+	return &loggerRegistry{
+		loggers: make(map[string]Logger),
+	}
+}
+
+func (lr *loggerRegistry) registerLogger(name string, logger Logger) {
+	lr.mu.Lock()
+	defer lr.mu.Unlock()
+	lr.loggers[name] = logger
+}
+
+func (lr *loggerRegistry) loggerNamed(name string) (logger Logger, ok bool) {
+	lr.mu.RLock()
+	defer lr.mu.RUnlock()
+	logger, ok = lr.loggers[name]
+	return logger, ok
+}
+
+func (lr *loggerRegistry) updateLoggerLevel(name string, level Level) error {
+	lr.mu.RLock()
```

Comment:
This should take a write lock, no?

Reply:
I think we are only reading from the map, but write to the log level which I believe is already atomic

Reply:
Yeah fair point; ignore my comment. I suppose this is no different from calling `loggerNamed` and then calling `logger.SetLevel` on the returned logger.